build.gradle

```diff
@@ -203,6 +203,10 @@ task unitTest(type: Test) {
 }
 
 task connectedTest(type: Test) {
+	environment = [
+			'TEST_ENV': System.getenv("TEST_ENV"),
+			'GOOGLE_APPLICATION_CREDENTIALS': "${googleCredentialsFile}"
```

Comment:
When I remove that it gives the following error:

```
Caused by: java.io.IOException: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
```

Reply:
Ah, what's probably happening is that L182 is reassigning the entire "environment" variable, which is a Map<String, Object> ([docs](https://docs.gradle.org/3.3/dsl/org.gradle.api.tasks.testing.Test.html#org.gradle.api.tasks.testing.Test:environment)).

Could you try changing that reassignment into a map addition? Maybe then the TEST_ENV will propagate without any explicit effort on our part...

L182:
```
environment.put('GOOGLE_APPLICATION_CREDENTIALS', "${googleCredentialsFile}");
```

Reply:
Adding it there didn't work - it was still being overwritten, but adding `environment.put('TEST_ENV'...)` to the connected and integration tasks worked!